### PR TITLE
Solr Search for WordPress - First Pass

### LINKED ISSUE #1291: Apache Solr on Pantheon Doc Update (WordPress)
Re: [Apache Solr on Pantheon](https://pantheon.io/docs/articles/sites/apache-solr)
## Issue Description:
WordPress Solr works and is on the repo. We need to document it as well as we have the Drupal implementation. 
## Suggested Resolution 
- [x] Decision: Make this article only about Solr containers running on Pantheon and move Drupal/Wordpress installs to separate article OR Add comprehensive WordPress install/usage instructions to Parallel Drupal ones. 
- [X] Add solr to questions-wp
- [x] Document setup/configuration
- [x] Create example query to extend (https://github.com/pantheon-systems/solr-power/issues/48)

### PR DESCRIPTION
**Pull request by rachelwhitton**
Closes #1291 

- Separate Drupal and WP docs for installation
- Keep general/platform/service info on main page

@bmackinney can you review and provide feedback/guidance? Not sure if splitting the doc based on CMS is ideal or not. ping @nataliejeremy 

This is a WIP - I wasn't able to search custom post types (tested with anspress q&a). I was able to search successfully on a new installation

### CODE REVIEW

File: source/docs/articles/sites/apache-solr.md [hunk 1]
```diff
@@ -12,101 +11,18 @@ Apache Solr is a system for indexing and searching site content. Pantheon provid
```
Comment (rachelwhitton) [inline source/docs/articles/sites/apache-solr.md]: @bmackinney the remaining content here is assumed to be true for both Drupal and WordPress. Can you confirm? Also, I'm not sure we need an agnostic doc to summarize Solr on Pantheon, but defer to you. I think we can probably move the remnants of this page to the Drupal doc and reiterate the WP doc alongside development of the plugin.

File: source/docs/articles/wordpress/solr.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ - Delete All
+ - Repost schema.xml
+ - Load All Post(s)
+ - Load All Page(s)
+ - Load All Attachment(s)
+ <div class="alert alert-info">
+ <h4>Note</h4>
+ To use a custom <code>schema.xml</code>, upload it to the <code>/wp-content/uploads/solr-for-wordpress-on-pantheon/</code> directory.
+ </div>
+6. Use the **Query** tab to search for words or phrases to validate Solr's indexing configuration.
```
Comment (rachelwhitton) [inline source/docs/articles/wordpress/solr.md]: We may want to touch on custom post types and include any known limitations in our first iteration. Let me know what you think @bmackinney

**Comment by nataliejeremy** [comment]
This one looks good to me. +1

File: source/docs/articles/wordpress/solr.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+---
+title: Enabling Solr for WordPress
+description: Detailed information on using Apache Solr with WordPress.
+category:
+    - wordpress
+    - developing
+keywords: apche, apache solr, index, indexing, searching, index and search, indexing and searching, solr, how to enable solr, enable solr, solr api, indexserver solr, solr indexserver, solr api, enable solr search, apachesolr, apache solor search module, solr modules, configure solr,
+---
+[Apache Solr](/docs/articles/sites/apache-solr) is a system for indexing and searching site content. First, you will need to add the Index Server to your site. From your Dashboard, go to **Settings** > **Add Ons** > **Apache Solr Index Server: Add**.
```
Comment (bmackinney) [inline source/docs/articles/wordpress/solr.md]: Consider noting this is not for Personal plan sites.

File: source/docs/articles/wordpress/solr.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ <div class="alert alert-info">
+ <h4>Note</h4> You can exclude pages or posts from being indexed by providing the numeric ID of the item (comma separated).
+ </div>
+5. Select the **Actions** tab and index your site's posts, pages, and/or attachments on Solr by clicking all that apply:
+ - **Load All post(s)**
+ - **Load All page(s)**
+ - **Load All attachment(s)**
+6. Use the **Query** tab to search for words and/or phrases to validate Solr's indexing configuration.
+
+Deploy the plugin to the site's Live environment after validation and testing. The `schema.xml` file must be resent to Solr when deploying the plugin to another environment. Select **Repost schema.xml** from the **Actions** tab, then validate expected results.
```
Comment (bmackinney) [inline source/docs/articles/wordpress/solr.md]: typo: `resent`

File: source/docs/articles/wordpress/solr.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+1. [Set the connection mode to SFTP](/docs/articles/sites/code/developing-directly-with-sftp-mode) for the Dev or Multidev environment via the Pantheon Dashboard or with [Terminus](/docs/articles/local/cli/):
+ ```bash
+ terminus site set-connection-mode --site=<site> --env=dev --mode=sftp
+ ```
+
+2. Install and activate the [Solr Search for WordPress (Solr Power)](https://wordpress.org/plugins/solr-power/) plugin on the Dev or Multidev environment using the WordPress Dashboard or with Terminus:
+ ```bash
+ terminus wp 'plugin install --activate solr-power' --site=<site> --env=dev
+ ```
+ Upon activation, the plugin will generate and send a `schema.xml` document to the Solr server.
```
Comment (bmackinney) [inline source/docs/articles/wordpress/solr.md]: Consider linking to the schema.xml file: https://github.com/pantheon-systems/solr-power/blob/master/schema.xml

File: source/docs/articles/wordpress/solr.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+6. Use the **Query** tab to search for words and/or phrases to validate Solr's indexing configuration.
+
+Deploy the plugin to the site's Live environment after validation and testing. The `schema.xml` file must be resent to Solr when deploying the plugin to another environment. Select **Repost schema.xml** from the **Actions** tab, then validate expected results.
+### Optimize Index
+For details, see [Optimize command](http://solarium.readthedocs.org/en/stable/queries/update-query/building-an-update-query/optimize-command/).
+
+## Known Issues/Limitations
+This plugin is under active development on [GitHub](https://github.com/pantheon-systems/solr-power). Use the issue queue for status updates and support. Pull requests are also welcome!
+### Automatic Re-Indexing
+Solr is updated automatically when a post is modified, published, or deleted. If modifications to the indexing options have been saved within the **Indexing** tab, the site must be indexed again within the **Actions** tab.
```
Comment (bmackinney) [inline source/docs/articles/wordpress/solr.md]: This seems unclear to me. Consider changing to, 

> ## Re-Indexing
New, deleted, and modified posts and pages are automatically added to the Solr index. However, if you modify the indexing options at `/wp-admin/path-to-options/`, you must manually re-index the site.

File: source/docs/articles/wordpress/solr.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+### Deploy to Test and Live
+The `schema.xml` file is not automatically sent to Solr when deploying from one environment to another. Once you have deployed to the Test and/or Live environment, log into the environment's URL and navigate to **Settings** > **Solr Options** > **Actions**. Select **Repost schema.xml** and re-index your site by selecting all that apply:
```
Comment (bmackinney) [inline source/docs/articles/wordpress/solr.md]: Clarify that this is only in respect to custom or changed custom `schema.xml` files on deployment, not for every deployment.

**Pull request merged**